src/display_controller.py

```diff
@@ -1005,6 +1006,10 @@ def run(self):
             return
              
         try:
+            self.cache_manager.clear_cache()
+            self._update_modules()
+            time.sleep(5)
+            self.current_display_mode = self.available_modes[self.current_mode_index] if self.available_modes else 'none'
```

Comment:
I added this to help with stale data or changes to the caching system, if the program restarts it will clear the cache and refetch everything before starting the loop, that way it will hopefully prevent any issues like with the "legacy" cache stuff u had to add.